src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -239,12 +239,42 @@ public static void Copy(Array sourceArray, long sourceIndex, Array destinationAr
             return InternalGetValue(GetFlattenedIndex(new ReadOnlySpan<int>(indices)));
         }
 
-        public object? GetValue(int index)
+        public unsafe object? GetValue(int index)
         {
-            if (Rank != 1)
+            MethodTable* pMethodTable = RuntimeHelpers.GetMethodTable(this);
+
+            // The array rank is 0 for SZ arrays, throw if that's not the case
+            if (pMethodTable->MultiDimensionalArrayRank != 0)
```

Comment:
This file is shared with all runtimes. CoreCLR specific details won't work here.

Reply:
Yup, just noticed this and @SingleAccretion was walking me through the process, I'll move this to the CoreCLR path and ifdef this out on CoreCLR 😅